### PR TITLE
h5utils: Fix build and add new version

### PR DESCRIPTION
**Pull request by bernhardkaindl**
@1.12.1+png depends_on libpng@:1.5.0 because it needs the old API

### CODE REVIEW

File: var/spack/repos/builtin/packages/h5utils/package.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ class H5utils(AutotoolsPackage):
     url      = "http://ab-initio.mit.edu/h5utils/h5utils-1.12.1.tar.gz"
     list_url = "http://ab-initio.mit.edu/h5utils/old/"
 
+    version('1.13.1', sha256='c5a76f064d6daa3e65583dce2b61202510e67cf6590f076af9a8aa72511d7d65')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/h5utils/package.py:16]: I cannot confirm the sha256.  I'm getting "All fetchers failed".

Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/h5utils/package.py:16]: @tldahlgren Fixed "spack checksum": Added new commit: make 'spack checksum' work for current and future versions

File: var/spack/repos/builtin/packages/h5utils/package.py [hunk 2]
```diff
@@ -12,10 +12,18 @@ class H5utils(AutotoolsPackage):
 
     homepage = "http://ab-initio.mit.edu/wiki/index.php/H5utils"
     url      = "http://ab-initio.mit.edu/h5utils/h5utils-1.12.1.tar.gz"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/h5utils/package.py]: You can have the main URL match the new download location using the following:

```suggestion
    url      = "https://github.com/NanoComp/h5utils/releases/download/1.13.1/h5utils-1.13.1.tar.gz"
```

File: var/spack/repos/builtin/packages/h5utils/package.py [hunk 3]
```diff
@@ -12,10 +12,18 @@ class H5utils(AutotoolsPackage):
...
 
+    version('1.13.1', sha256='c5a76f064d6daa3e65583dce2b61202510e67cf6590f076af9a8aa72511d7d65')
     version('1.12.1', sha256='7290290ca5d5d4451d757a70c86baaa70d23a28edb09c951b6b77c22b924a38d')
 
+    def url_for_version(self, version):
+        """The release of 1.12.1 isn't on github but on ab-initio, using self.url"""
+        url = 'https://github.com/NanoComp/h5utils/releases/download/{0}/h5utils-{0}.tar.gz'
+        if (version == Version('1.12.1')):
+            return self.url
+        return url.format(version)
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/h5utils/package.py]: And use a secure location for v1.12.1 plus simplify the package with the above `url` fix PLUS:

```suggestion

    version('1.13.1', sha256='c5a76f064d6daa3e65583dce2b61202510e67cf6590f076af9a8aa72511d7d65')
    version('1.12.1', sha256='7290290ca5d5d4451d757a70c86baaa70d23a28edb09c951b6b77c22b924a38d',
            url = "https://github.com/NanoComp/h5utils/archive/refs/tags/1.12.1.tar.gz")
```

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/h5utils/package.py]: Otherwise, at least move the `url_for_version` method to the bottom of the file.

**Review by tldahlgren — CHANGES_REQUESTED** [review]

**Comment by bernhardkaindl** [comment]
I've accepted all changes and fixed the spack style. Ready for merge.

**Comment by bernhardkaindl** [comment]
@tldahlgren I merged your requested changed, can you confirm/approve? Thanks!

File: var/spack/repos/builtin/packages/h5utils/package.py [hunk 4]
```diff
@@ -11,10 +11,11 @@ class H5utils(AutotoolsPackage):
     scientific data in the free, portable HDF5 format."""
 
     homepage = "http://ab-initio.mit.edu/wiki/index.php/H5utils"
-    url      = "http://ab-initio.mit.edu/h5utils/h5utils-1.12.1.tar.gz"
-    list_url = "http://ab-initio.mit.edu/h5utils/old/"
+    url      = "https://github.com/NanoComp/h5utils/releases/download/1.13.1/h5utils-1.13.1.tar.gz"
 
-    version('1.12.1', sha256='7290290ca5d5d4451d757a70c86baaa70d23a28edb09c951b6b77c22b924a38d')
+    version('1.13.1', sha256='c5a76f064d6daa3e65583dce2b61202510e67cf6590f076af9a8aa72511d7d65')
+    version('1.12.1', sha256='7290290ca5d5d4451d757a70c86baaa70d23a28edb09c951b6b77c22b924a38d',
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/h5utils/package.py]:
```suggestion
    version('1.12.1', sha256='7e6db86fee00a8008f78b2be921177042c661203c0936b078fcc8f9c71e4a883',
```

File: var/spack/repos/builtin/packages/h5utils/package.py [hunk 5]
```diff
@@ -11,10 +11,11 @@ class H5utils(AutotoolsPackage):
     scientific data in the free, portable HDF5 format."""
 
     homepage = "http://ab-initio.mit.edu/wiki/index.php/H5utils"
-    url      = "http://ab-initio.mit.edu/h5utils/h5utils-1.12.1.tar.gz"
-    list_url = "http://ab-initio.mit.edu/h5utils/old/"
+    url      = "https://github.com/NanoComp/h5utils/releases/download/1.13.1/h5utils-1.13.1.tar.gz"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/h5utils/package.py:14]: Love the switch to `https`!

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Checked the sha256 values for the two versions.  One needs to be updated to reflect the new URL.

File: var/spack/repos/builtin/packages/h5utils/package.py (hunk 5 above)
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/h5utils/package.py:14]: This was replaced (and list_url was removed already), it was in an old commit.

Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/h5utils/package.py:14]: @tldahlgren your requested changes are applied.

**Comment by bernhardkaindl** [comment]
@tldahlgren Both requested changes are applied in the new package.py.

**Review by tldahlgren — APPROVED** [review]
Confirmed the sha256 (via `spack checksum`).

**Comment by bernhardkaindl** [comment]
@tldahlgren thanks!

**Pull request merged**